### PR TITLE
feat: add s3 schema

### PR DESCRIPTION
**Pull request by netanelC**

### CODE REVIEW

File: schemas/common/s3/full/v1.schema.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          "description": "The storage class to store newly created objects",
+          "default": "STANDARD",
+          "x-env-value": "S3_STORAGE_CLASS"
+        },
+        "maxAttempts": {
+          "type": "number",
+          "description": "The max attempts for executing S3 command",
+          "default": 3
+        }
+      }
```
Comment (CptSchnitz) [inline schemas/common/s3/full/v1.schema.json:34]: is it a setting we want for all the users of this bucket?

File: schemas/common/s3/partial/v3.schema.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      "description": "Whether to force path style URLs for S3",
+      "default": true,
+      "x-env-value": "S3_FORCE_PATH_STYLE"
+    },
+    "sslEnabled": {
+      "type": "boolean",
+      "description": "Whether SSL is enabled for requests",
+      "default": false,
+      "x-env-value": "S3_SSL_ENABLED"
+    },
```
Comment (CptSchnitz) [inline schemas/common/s3/partial/v3.schema.json:53]: do we need both http/https and ssl enabled?

**Review by CptSchnitz — CHANGES_REQUESTED** [review]
what about validations? (format/regex/length)

File: schemas/common/s3/full/v1.schema.json (hunk 1 above)
Comment (netanelC) [reply schemas/common/s3/full/v1.schema.json:34]: Removed

File: schemas/common/s3/partial/v3.schema.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      "type": "string",
+      "description": "The access key id of the bucket",
+      "x-env-value": "S3_ACCESS_KEY"
+    },
+    "secretAccessKey": {
+      "type": "string",
+      "description": "The secret access key of the bucket",
+      "x-env-value": "S3_SECRET_KEY"
+    },
+    "host": {
```
Comment (ronenkapelian) [inline schemas/common/s3/partial/v3.schema.json:17]: it should be called "endpoint" instead "host"

```suggestion
    "endpoint": {
```

**Review by ronenkapelian — CHANGES_REQUESTED** [review]
please notice comments

File: schemas/common/s3/partial/v3.schema.json (hunk 3 above)
Comment (netanelC) [reply schemas/common/s3/partial/v3.schema.json]: Changed

**Review by ronenkapelian — APPROVED** [review]

File: schemas/common/s3/partial/v3.schema.json
Comment (CptSchnitz) [inline schemas/common/s3/partial/v3.schema.json:1]: missing description

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: schemas/common/s3/partial/v3.schema.json
Comment (netanelC) [reply schemas/common/s3/partial/v3.schema.json:1]: Added

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**